docs/ProductionChecklistForScalarDBCluster.md

```diff
@@ -0,0 +1,140 @@
+
+From the perspective of performance, we recommend using the `direct-kubernetes` mode. To use the `direct-kubernetes` mode, you must deploy your application pods on the same Kubernetes cluster as ScalarDB Cluster pods. In this case, you do not need to deploy Envoy pods.
+
+If you cannot deploy your Java application pods on the same Kubernetes cluster as ScalarDB Cluster pods for some reason, you must use the `indirect` mode. In this case, you must deploy Envoy pods.
+
+{% capture notice--info %}
+**Note**
+
+The client mode configuration is dedicated to the Java client library. When you use other programming languages than Java (i.e., you use [gRPC API](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/scalardb-cluster-grpc-api-guide.md) and [gRPC SQL API](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/scalardb-cluster-sql-grpc-api-guide.md) without the Java client library) for your application, there is no such configuration. In this case, you must deploy Envoy pods.
+{% endcapture %}
+
+<div class="notice--info">{{ notice--info | markdownify }}</div>
+
+### Transaction manager configuration (Java client library only)
+
+You must always access ScalarDB Cluster. To ensure requests are running properly, check the properties file for your client application and confirm that `scalar.db.transaction_manager=cluster` is configured when you use CRUD API.
+
+* Recommended for production environments
+  ```mermaid
+  flowchart LR
+    A(<b>App</b> <br> ScalarDB Cluster Library with gRPC) --> B(<b>ScalarDB Cluster</b> <br> ScalarDB Library with Consensus Commit) --> C(Underlying storage/database)
+  ```
+
+* Not recommended for production environments (for testing purposes only)
+  ```mermaid
+  flowchart LR
+    A(<b>App</b> <br> ScalarDB Cluster Library with Consensus Commit) --> B(Underlying storage/database)
+  ```
+
+### SQL connection configuration (Java client library only)
+
+You must always access ScalarDB Cluster. To ensure requests are running properly, check the properties file for your client application and confirm that `scalar.db.sql.connection_mode=cluster` is configured when you use SQL API.
+
+* Recommended for production environments
+  ```mermaid
+  flowchart LR
+    A("<b>App</b> <br> ScalarDB SQL Library (Cluster mode)") --> B(<b>ScalarDB Cluster</b> <br> ScalarDB Library with Consensus Commit) --> C(Underlying storage/database)
+  ```
+
+* Not recommended for production environments (for testing purposes only)
+  ```mermaid
+  flowchart LR
+    A("<b>App</b> <br> ScalarDB SQL Library (Direct mode)") --> B(Underlying storage/database)
+  ```
```

Comment:
This suggestion slightly modifies the shapes in the Mermaid diagrams and replaces bullets with headings.

````suggestion
#### Recommended for production environments

```mermaid
flowchart LR
  app["<b>App</b><br />ScalarDB SQL Library (Cluster mode)"]
  server["<b>ScalarDB Cluster</b><br />ScalarDB Library with<br />Consensus Commit"]
  db[(Underlying storage or database)]
  app --> server --> db
```

#### Not recommended for production environments (for testing purposes only)

```mermaid
flowchart LR
  app["<b>App</b><br />ScalarDB SQL Library (Direct mode)"]
  db[(Underlying storage or database)]
  app --> db
```
````